tests/cxx/unit_tests/ghostfragment/drivers/fragment.cpp

```diff
@@ -1,78 +1,64 @@
-// #include "../test_ghostfragment.hpp"
+#include "../test_ghostfragment.hpp"
+#include "hydrocarbon/hydrocarbon.h"
+#include <ghostfragment/property_types/fragmented_nuclei.hpp>
 
-// using namespace ghostfragment;
-// using namespace testing;
+using namespace ghostfragment;
+using namespace testing;
 
-// using frags_pt       = simde::FragmentedMolecule;
-// using graph_pt       = pt::MolecularGraph;
-// using graph2frags_pt = pt::MolecularGraphToFragments;
+using frags_pt       = pt::FragmentedNuclei;
+using graph_pt       = pt::MolecularGraph;
+using graph2frags_pt = pt::MolecularGraphToFragments;
 
-// using frags_type = type::fragmented_molecule;
-// using graph_type = MolecularGraph;
-// using conns_type = graph_type::connectivity_type;
-// using mol_type   = typename frags_type::superset_type;
-// namespace {
+using frags_type = chemist::FragmentedNuclei;
+using graph_type = MolecularGraph;
+using conns_type = graph_type::connectivity_type;
+using mol_type   = chemist::Molecule;
 
-// auto make_patom_module(const mol_type& mol, const frags_type& rv) {
-//     return pluginplay::make_lambda<frags_pt>([=](auto&& mol_in) {
-//         REQUIRE(mol_in == mol);
-//         return rv;
-//     });
-// }
+/* Testing strategy:
+ *
+ * The FragmentDriver module is purely a driver. If we assume the modules it
+ * calls work correctly, the only thing we need to test is that the data flows.
+ * Testing is done using the default modules (Heavy Atom for pseudoatom generation,
+ * Bond-Based for fragmentation).
+ */
```

Comment:
For an integration test that's okay (so if you want to create a directory `tests/cxx/integration_tests/ghostfragment/drivers` and move this file there that's fine), but for unit tests we really want to be testing this module in isolation. This in turn means we need to re-enable the lambda modules you commented out.

To clarify, unit tests try to test a code component in relative isolation. Meaning we want the test to have minimal additional dependencies. The goal of a unit test is really to try and catch any sort of problem intrinsic with the code we are testing. 

Integration tests couple several pieces together and make sure they work smoothly together. Unlike unit tests, integration tests can fail if any of the pieces break; however, if one has unit tests for each piece, it's also likely that the unit test for that piece is also failing. Ultimately, the goal of integration tests are to make sure there's no surprises when you start putting the pieces together.

Reply:
Added integration test, changed unit test to use lambda modules.